src/main/java/org/gitlab4j/api/DiscussionsApi.java

```diff
@@ -671,4 +671,86 @@ public void deleteMergeRequestThreadNote(Object projectIdOrPath, Integer mergeRe
+
+    /**
+     * Adds a new note to the thread.
+     *
+     * <pre><code>GitLab Endpoint: POST /projects/:id/issues/:issue_iid/discussions/:discussion_id/notes</code></pre>
+     *
+     * @param projectIdOrPath projectIdOrPath the project in the form of an Integer(ID), String(path), or Project instance
+     * @param issueIid The IID of an issue
+     * @param discussionId the id of discussion
+     * @param body the content of the note
+     * @param createdAt (optional) date the discussion was created (requires admin or project/group owner rights)
+     * @return a Note instance containing the newly created note
+     * @throws GitLabApiException if any exception occurs during execution
+     */
+    public Note addIssueThreadNote(Object projectIdOrPath, Integer issueIid,
+                                   String discussionId, String body, Date createdAt) throws GitLabApiException {
+        GitLabApiForm formData = new GitLabApiForm()
+                .withParam("body", body, true)
+                .withParam("created_at", createdAt);
+        Response response = post(Response.Status.CREATED, formData,
+                "projects", getProjectIdOrPath(projectIdOrPath), "issues", issueIid, "discussions", discussionId, "notes");
+        return (response.readEntity(Note.class));
+    }
+
+    /**
+     * Modify existing thread note of an issue.
+     *
+     * <pre><code>GitLab Endpoint: PUT /projects/:id/issues/:issue_iid/discussions/:discussion_id/notes/:note_id</code></pre>
+     *
+     * @param projectIdOrPath projectIdOrPath the project in the form of an Integer(ID), String(path), or Project instance
+     * @param issueIid The IID of an issue
+     * @param discussionId the id of discussion
+     * @param noteId the id of the note
+     * @param body the content of the note
+     * @return a Note instance containing the modified note
+     * @throws GitLabApiException if any exception occurs during execution
+     */
+    public Note modifyIssueThreadNote(Object projectIdOrPath, Integer issueIid,
+                                      String discussionId, Integer noteId, String body) throws GitLabApiException {
+        GitLabApiForm formData = new GitLabApiForm()
+                .withParam("body", body, true);
+        Response response = putWithFormData(Response.Status.OK, formData,
+                "projects", getProjectIdOrPath(projectIdOrPath), "issues", issueIid, "discussions", discussionId, "notes", noteId);
+        return (response.readEntity(Note.class));
+    }
+
+    /**
+     * Deletes an existing thread note of an issue.
+     *
+     * <pre><code>GitLab Endpoint: DELETE /projects/:id/issues/:issue_iid/discussions/:discussion_id/notes/:note_id</code></pre>
+     *
+     * @param projectIdOrPath projectIdOrPath the project in the form of an Integer(ID), String(path), or Project instance
+     * @param issueIid The IID of an issue
+     * @param discussionId the id of discussion
+     * @param noteId the id of the note
```

Comment:
Need to add the following to the Javadocs:
```
* @throws GitLabApiException if any exception occurs during execution
```

Reply:
added the javadoc for the exception